src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx

```diff
@@ -15,43 +24,189 @@ const ContactTagChip = styled(Chip)(({ theme }) => ({
+              enqueueSnackbar(t('Tag successfully removed'), {
+                variant: 'success',
+              });
+            }
+          },
+        });
+      } catch (error) {
+        enqueueSnackbar(error.message, { variant: 'error' });
+        throw error;
+      }
     }
-    console.info('You clicked the Delete tag for tag' + tag + '' + contactId);
   };
-  const handleAddTag = () => {
-    // TODO: add mutation to update tag for contact
-    console.info('You clicked add contact Tag to contact' + contactId);
+
+  const onSubmit = async (
+    { tag }: { tag: string },
+    { resetForm }: FormikHelpers<{ tag: string }>,
+  ): Promise<void> => {
+    try {
+      resetForm();
+      updateContactTags({
+        variables: {
+          accountListId,
+          contactId,
+          tagList: [...contactTags, tag],
+        },
+        optimisticResponse: {
+          updateContact: {
+            contact: {
+              id: contactId,
+              tagList: [...contactTags, tag],
+            },
+          },
+        },
+        update: (cache, { data: updateData }) => {
+          const tagList = updateData?.updateContact?.contact.tagList;
+          const query = {
+            query: ContactDetailsTabDocument,
+            variables: {
+              accountListId,
+              contactId,
+            },
+          };
+          const dataFromCache = cache.readQuery<ContactDetailsTabQuery>(query);
+
+          if (dataFromCache) {
+            const data = {
+              contact: {
+                ...dataFromCache.contact,
+                tagList,
+              },
+            };
+            cache.writeQuery({ ...query, data });
+            enqueueSnackbar(t('Tag successfully added'), {
+              variant: 'success',
+            });
+          }
+        },
+      });
+    } catch (error) {
+      enqueueSnackbar(error.message, { variant: 'error' });
+      throw error;
+    }
   };
 
   return (
-    <ContactTagsContainer>
+    <ContactTagsContainer display="flex" alignItems="center">
+      <ContactTagIcon />
       {contactTags.map((tag) => (
-        <ContactTagChip key={tag} label={tag} onDelete={handleTagDelete} />
+        <ContactTagChip
+          key={tag}
+          label={tag}
+          disabled={updating}
+          onDelete={() => handleTagDelete(tag)}
+          title={t('Delete Icon')}
+        />
       ))}
-      <Link color="textSecondary" onClick={handleAddTag}>
-        <ContactTagAddTagText variant="subtitle1">
-          {t('add tag')}
-        </ContactTagAddTagText>
+      <Link color="textSecondary">
+        <Formik
+          initialValues={{ tag: '' }}
+          validationSchema={tagSchema}
+          onSubmit={onSubmit}
+        >
+          {({
+            values: { tag },
+            handleChange,
+            handleSubmit,
+            isSubmitting,
+          }): ReactElement => (
+            <form onSubmit={handleSubmit} noValidate>
+              <ContactTagInput
```

Comment:
I think this is going to need some kind of autocomplete dropdown. Seems like in the existing app you can make up new arbitrary tags and then they are suggested when you start typing them on other contacts. Idk if it should be a new PR or not.

Reply:
I think I'm going to merge this in and make a follow-up ticket to add the autocomplete.